components/upload/UploadList.tsx

```diff
@@ -170,13 +172,21 @@ export default class UploadList extends React.Component<UploadListProps, any> {
       const linkProps =
         typeof file.linkProps === 'string' ? JSON.parse(file.linkProps) : file.linkProps;
 
-      const removeIcon = showRemoveIcon ? (
+      const removeIcon = showRemoveIcon ? customRemoveIcon && (
+        <span title={locale.removeFile} onClick={() => this.handleClose(file)}>
+          {customRemoveIcon}
+        </span>
```

Comment:
React.cloneElement(customRemoveIcon, ?

Reply:
I wil try React.cloneElement

Reply:
A problem: If someone pass an element with onClick, should I overwrite or keep its own onClick? @yoyo837

Reply:
Another problem: If someone pass a string type element to removeIcon, React.cloneElement cannot deal with string while <span> can. Should I seperate these two scenes or simply keep <span>?

Reply:
> Should I seperate these two scenes or simply keep span?

 Two scenes with `React.isValidElement`

> If someone pass an element with onClick, should I overwrite or keep its own onClick

Call `handleClose` then call its own `onClick`, I think.